x-pack/plugins/apm/public/components/shared/time_comparison/index.tsx

```diff
@@ -31,19 +36,58 @@ export function TimeComparison() {
   const history = useHistory();
   const { isSmall } = useBreakpoints();
   const {
-    query: { rangeFrom, rangeTo, comparisonEnabled, offset },
+    query: { rangeFrom, rangeTo, comparisonEnabled, offset, environment },
   } = useAnyOfApmParams('/services', '/backends/*', '/services/{serviceName}');
 
+  const { anomalyDetectionJobsStatus, anomalyDetectionJobsData } =
+    useAnomalyDetectionJobsContext();
+  const { core } = useApmPluginContext();
+
   const { start, end } = useTimeRange({ rangeFrom, rangeTo });
 
-  const comparisonOptions = getComparisonOptions({ start, end });
+  const canGetJobs = !!core.application.capabilities.ml?.canGetJobs;
+  const comparisonOptions = useMemo(() => {
+    const timeComparisonOptions = getComparisonOptions({ start, end });
+
+    const hasMLJobsMatchingEnv =
+      canGetJobs &&
+      anomalyDetectionJobsStatus === 'success' &&
+      Array.isArray(anomalyDetectionJobsData?.jobs) &&
+      anomalyDetectionJobsData?.jobs.some((j) => j.environment === environment);
+
+    if (canGetJobs) {
+      timeComparisonOptions.push({
+        value: TimeRangeComparisonEnum.ExpectedBounds,
+        text: i18n.translate('xpack.apm.comparison.mlExpectedBounds', {
+          defaultMessage: 'Expected bounds',
+        }),
+        disabled: !hasMLJobsMatchingEnv,
+      });
```

Comment:
1. Can we move this logic inside `getComparisonOptions`? 
2. We should not disable the option until the job has loaded. Right now if I select "Expected bounds" and then refresh the page I'm redirected back to one of the other options because "Expected bounds" is initially disabled. 
3. @dgieselaar You did a workaround so we'd show ML results even when "All" option was selected, if there was only one environment. Is the logic sharable or should it be re-implemented here?

Reply:
For the last question I found where we implement the logic: https://github.com/elastic/kibana/blob/a807c90310d2adf5c43694a8e4ae1f982ade0e32/x-pack/plugins/apm/common/anomaly_detection/get_preferred_service_anomaly_timeseries.ts#L30-L33

Reply:
Suggestion: we add the `preferedEnvironment` logic to the [environmentContext](https://github.com/elastic/kibana/blob/a807c90310d2adf5c43694a8e4ae1f982ade0e32/x-pack/plugins/apm/public/context/environments_context/environments_context.tsx).

Then usage changed like:

```diff
- const { environment, environments } = useEnvironmentsContext();
+ const { preferedEnvironment, environment, environments } = useEnvironmentsContext();
```

Reply:
Updated here https://github.com/elastic/kibana/pull/132456/commits/11e2b8dd0ad5dd7b2aa0b1cb8566189a259ad96b